angularjs-portal-home/src/main/webapp/my-app/layout/widget/controllers.js

```diff
@@ -69,6 +84,21 @@ define(['angular'], function(angular){
         });
       }
     };
+    
+    $scope.changePref=function(userPreference){
+
+    	if(userPreference==='C'){
+    		$scope.changeToC();
+    		$scope.showMetric=true;
+    	}else{
+    		$scope.changeToF();
+    	}
+
+    	var value = {};
+    	value.userWeatherPreference = userPreference;
+       	keyValueService.setValue($scope.fetchKey, value);
```

Comment:
So on initial setting that you do on line 75, wouldn't this then immediately save? We would want to avoid this. Init flag param?